packages/core/src/fhir-deduplication/shared.ts

```diff
@@ -173,9 +175,12 @@ export function getDateFromResource<T extends Resource>(
   } else if ("period" in resource) {
     const period = resource.period;
     if (period.start) return getDateFromString(period.start, dateFormat);
+    else if (period.end) return getDateFromString(period.end, dateFormat);
   } else if ("effectivePeriod" in resource) {
     if (resource.effectivePeriod.start) {
-      return getDateFromString(resource.effectivePeriod.start, dateFormat);
+      return getDateFromString(resource.effectivePeriod.start);
```

Comment:
i thought we've already implemented this code. I rmb we discussed it

Reply:
adding date format - should help with diagnostic Report